hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/insights/om/om.tsx

```diff
@@ -777,6 +820,66 @@ export class Om extends React.Component<Record<string, object>, IOmdbInsightsSta
     });
   };
 
+  // Pending Delete Directories
+  fetchDeletePendingDir = (limit: number, prevKeyDirDelete: string) => {
+    this.setState({
+      loading: true,
+      nextClickable: true,
+      prevClickable: true
+    });
+
+    //Cancel any previous pending request
+    cancelRequests([
+      cancelMismatchedEndpointSignal,
+      cancelOpenKeysSignal,
+      cancelDeletePendingSignal,
+      cancelDeletedKeysSignal,
+      cancelRowExpandSignal,
+      cancelDeletedPendingDirSignal
+    ]);
+
+    let deletePendingDirEndpoint;
+    if (prevKeyDirDelete === "" || prevKeyDirDelete === undefined ) {
+      deletePendingDirEndpoint = `/api/v1/keys/deletePending/dirs?limit=${limit}`;
+    }
+    else {
+      deletePendingDirEndpoint = `/api/v1/keys/deletePending/dirs?limit=${limit}&prevKey=${prevKeyDirDelete}`;
+    }
+
+    const { request, controller } = AxiosGetHelper(deletePendingDirEndpoint, cancelDeletedPendingDirSignal);
+    cancelDeletedPendingDirSignal = controller 
+    request.then(deletePendingDirResponse => {
+      const deletedDirInfo = deletePendingDirResponse && deletePendingDirResponse.data && deletePendingDirResponse.data.deletedDirInfo;
```

Comment:
Store the value of `deletePendingDirResponse.data.lastKey` into another constant called `lastKey` and use it everywhere instead of accessing it again and again from the the response. 
```    
const lastKey = deletePendingDirResponse?.data?.lastKey || "";
```

Reply:
deletePendingDirResponse?.data?.lastKey is advance and babel is not supporting but need to change logic once this pr gets merged. I will do once PR got merged need to change pagination and other logic when we are going on last page.

https://github.com/apache/ozone/pull/5658